### ISSUE TITLE
st-gcn 1.0 train accuracy problem

### ISSUE THREAD

**Issue by ttthappy** [zhujiagang/st-gcn-data-len#1]
@zhujiagang thank you for sharing your experiences at https://github.com/yysijie/st-gcn/issues/23.  I want to train the st-gcn model in NTU RGB+D dataset. I use the old version code with default parameters. But, after more 80 epoch's training I got poor result. Then I modify the learning rate according to your paper, but the accuracy is still poor（Top1: 1.67%, Top5: 8.27%）. As a new student in this field, I was puzzled if I missed some details. Hope your reply. Thanks a lot.

[ Sat Dec 22 18:59:51 2018 ] Training epoch: 80
[ Sat Dec 22 19:00:03 2018 ] 	Batch(0/589) done. Loss: 4.1821  lr:0.001000
[ Sat Dec 22 19:03:31 2018 ] 	Batch(100/589) done. Loss: 4.1529  lr:0.001000
[ Sat Dec 22 19:06:57 2018 ] 	Batch(200/589) done. Loss: 4.1635  lr:0.001000
[ Sat Dec 22 19:10:27 2018 ] 	Batch(300/589) done. Loss: 4.1190  lr:0.001000
[ Sat Dec 22 19:13:55 2018 ] 	Batch(400/589) done. Loss: 4.1429  lr:0.001000
[ Sat Dec 22 19:17:22 2018 ] 	Batch(500/589) done. Loss: 4.1611  lr:0.001000
[ Sat Dec 22 19:20:24 2018 ] 	Mean training loss: 4.1409.
[ Sat Dec 22 19:20:24 2018 ] 	Time consumption: [Data]01%, [Network]99%
[ Sat Dec 22 19:20:24 2018 ] Eval epoch: 80
[ Sat Dec 22 19:24:07 2018 ] 	Mean test loss of 296 batches: 4.106204550008516.
[ Sat Dec 22 19:24:08 2018 ] 	Top1: 1.26%
[ Sat Dec 22 19:24:08 2018 ] 	Top5: 9.44%

modify the learning rate according to your paper:

[ Tue Dec 25 06:20:27 2018 ] Training epoch: 20
[ Tue Dec 25 06:20:48 2018 ] 	Batch(0/589) done. Loss: 4.5612  lr:0.001000
[ Tue Dec 25 06:25:36 2018 ] 	Batch(100/589) done. Loss: 4.8982  lr:0.001000
[ Tue Dec 25 06:30:21 2018 ] 	Batch(200/589) done. Loss: 4.5673  lr:0.001000
[ Tue Dec 25 06:35:04 2018 ] 	Batch(300/589) done. Loss: 4.6968  lr:0.001000
[ Tue Dec 25 06:39:46 2018 ] 	Batch(400/589) done. Loss: 4.5363  lr:0.001000
[ Tue Dec 25 06:44:30 2018 ] 	Batch(500/589) done. Loss: 4.4839  lr:0.001000
[ Tue Dec 25 06:48:41 2018 ] 	Mean training loss: 4.6989.
[ Tue Dec 25 06:48:41 2018 ] 	Time consumption: [Data]01%, [Network]99%
[ Tue Dec 25 06:48:41 2018 ] Eval epoch: 20
[ Tue Dec 25 06:53:54 2018 ] 	Mean test loss of 296 batches: 4.28557998747439.
[ Tue Dec 25 06:53:54 2018 ] 	Top1: 1.52%
[ Tue Dec 25 06:53:55 2018 ] 	Top5: 8.95%

**Comment by zhujiagang** [comment]
You should probably normalize the skeleton by substracting the frame center or starting frame center as the issue said.

**Comment by ttthappy** [comment]
Thank you very much!

**Comment by imduygusarikaya** [comment]
@ttthappy @zhujiagang I want to apply your method to my custom dataset with fewer interest points and junctions. I have never worked with skeletons before, can you advise me how I can do the normalization? Is there any framework for this?

**Comment by happysheep224** [comment]
I met the same problem, could you please tell me how you fixed it?
